src/GraphQL.AotCompilationSample/ServiceCollection.cs

```diff
@@ -0,0 +1,234 @@
+using System.Collections;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq.Expressions;
+using System.Reflection;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace GraphQL.AotCompilationSample;
+
+internal interface IAotServiceProviderConfiguration
+{
+    IAotServiceProviderConfiguration AddListType<T>();
+}
+
+internal class AotServiceProvider : IServiceProvider, IDisposable, IServiceScopeFactory, IServiceScope, IAotServiceProviderConfiguration
```

Comment:
I'm thinking I'll move this class to a NuGet package - perhaps Shane32.AotServiceProvider